src/search/csearch.c

```
@@ -16,19 +16,65 @@ version 0.1
+static const char *sr_consume_arg(int argc, char *argv[], int *i_arg,
+                                  const char *missing_msg)
+{
+  (*i_arg)++;
+  if (*i_arg < argc) return argv[*i_arg];
+#ifdef ERROR
+  fprintf(STDERR,"%s", missing_msg);
+#endif
+  exit(1);
+}
+
+static int sr_parse_positive_int(const char *value, const char *invalid_msg)
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'sr_parse_positive_int' of similar type ('const char *') are easily swapped by mistake

(bugprone-easily-swappable-parameters)